### PR TITLE
fail CI if events library is a dependency and outdated

### PR DESCRIPTION
**Pull request by vxio**

### CODE REVIEW

File: go/lint-project.sh [hunk 1]
```diff
@@ -51,6 +51,17 @@ then
     fi
 fi
 
+# Verify we're using the latest version of github.com/moovfinancial/events if it's a dependency
+eventsLibrary="github.com/moovfinancial/events"
```
Comment (adamdecaf) [inline go/lint-project.sh:55]: Can we guard this check to only run in moovfinancial projects? Similar to what https://github.com/moov-io/infra/pull/162/files did.

Comment (vxio) [reply go/lint-project.sh:55]: Good idea

File: go/lint-project.sh [hunk 2]
```diff
@@ -51,6 +51,17 @@ then
...
+eventsVersion=$( go list -u -m -mod=mod $eventsLibrary | awk -F'[][]' '{print $2}')
+if [[ $eventsVersion ]]
+then
+    echo "$eventsLibrary needs to be updated to the latest release: $eventsVersion"
```
Comment (adamdecaf) [inline go/lint-project.sh:59]:
```suggestion
    echo "$eventsLibrary needs to be updated to the latest release: $eventsVersion" 
    echo "Run 'go get -u ""$eventsLibrary""@latest' to resolve this issue"
```

I think this is the right syntax to properly generate a command they can run.

Comment (vxio) [reply go/lint-project.sh:59]: running this outputs
```bash
github.com/moovfinancial/events needs to be updated to the latest release: v0.4.12
Run 'go get -u github.com/moovfinancial/events@latest' to resolve this issue
```

Comment (vxio) [reply go/lint-project.sh]: I'll update this to try to get it to send a link

**Review by adamdecaf — APPROVED** [review]

**Pull request merged**